app/helpers/collection_time_helper.rb

```diff
@@ -1,16 +1,26 @@
 module CollectionTimeHelper
+  def collection_start_year(date)
+    window_end_date = Time.zone.local(date.year, 4, 1)
+    date < window_end_date ? date.year - 1 : date.year
+  end
+
   def current_collection_start_year
-    today = Time.zone.now
-    window_end_date = Time.zone.local(today.year, 4, 1)
-    today < window_end_date ? today.year - 1 : today.year
+    collection_start_year(Time.zone.now)
   end
 
   def collection_start_date(date)
-    window_end_date = Time.zone.local(date.year, 4, 1)
-    date < window_end_date ? Time.zone.local(date.year - 1, 4, 1) : Time.zone.local(date.year, 4, 1)
+    Time.zone.local(collection_start_year(date), 4, 1)
   end
 
   def current_collection_start_date
     Time.zone.local(current_collection_start_year, 4, 1)
   end
+
+  def collection_end_date(date)
+    Time.zone.local(collection_start_year(date) + 1, 3, 31)
```

Comment:
collection_start_year(date) + 1.year - 1.day, less hardcoded, still looks like it makes sense? Maybe personal preference, so no need to change if you disagree

Reply:
or +364.days 🫠

Reply:
Yes, I considered both those options, but then decided what I came up with was less confusing. This definitely should live in the form itself, as discussed!